python/lsst/sconsUtils/scripts.py

```diff
@@ -174,10 +177,11 @@ def finish(defaultTargets=DEFAULT_TARGETS,
         state.env.Requires(state.targets["python"], state.targets["version"])
         declarer = state.env.Declare()
         state.env.Requires(declarer, install)  # Ensure declaration fires after installation available
+
+        # Since we are translating targets to directory names, shebang must be translated
+        defaultTargets = utils.targetsToDirs(defaultTargets)
+
         state.env.Default([t for t in defaultTargets if os.path.exists(t)])
-        # shebang target is not named same as relevant directory so we must be explicit
-        if "shebang" in defaultTargets and os.path.exists("bin.src"):
-            state.env.Default("shebang")
```

Comment:
Are you sure that the original code here is not sufficient and that you want `state.env.Default("bin.src")` instead of `state.env.Default("shebang")`?  That looks fishy.

Reply:
That's what I did have originally and it wasn't working. It now works so I am guessing it really did need paths and not targets there. Changing this bit of code made all the difference.

Reply:
with code as is:
```
scons: Nothing to be done for `python'.
scons: `bin.src' is up to date.
rewrite_shebang(["bin/makeLinkFarm"], ["bin.src/makeLinkFarm"])
rewrite_shebang(["bin/sconsOpts"], ["bin.src/sconsOpts"])
```
with the old code:
```
rewrite_shebang(["bin/makeLinkFarm"], ["bin.src/makeLinkFarm"])
rewrite_shebang(["bin/sconsOpts"], ["bin.src/sconsOpts"])
scons: done building targets.
```
first time and:
```
scons: `shebang' is up to date.
```
So I think the problem is that `shebang` always ended up on the end of the list in the old version and was filtered out of the right place. We need `shebang` to turn up in the correct place even if there is no directory named for it.